File: domein/src/main/java/be/genealogie/domein/dto/RelatieDto.java
package be.genealogie.domein.dto;

import be.genealogie.domein.Geslacht;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class RelatieDto {
    private Long id;
    private NatuurlijkPersoonDTO persoon1;
    private NatuurlijkPersoonDTO persoon2;
    private LocalDate gehuwdOp;
    private String gehuwdTe;
    private List<NatuurlijkPersoonDTO> kinderen;

    public NatuurlijkPersoonDTO getMan() {
        return persoon1.getGeslacht() == Geslacht.MAN? persoon1 : persoon2;
    }

    public NatuurlijkPersoonDTO getVrouw() {
        return persoon1.getGeslacht() == Geslacht.VROUW? persoon1 : persoon2;
    }
}

File: frontend/src/app/model/relatie-dto.ts
import {NatuurlijkPersoonDto} from "./natuurlijk-persoon-dto";

export class RelatieDto {
  public constructor(public id: number,
                     public persoon1: NatuurlijkPersoonDto,
                     public persoon2: NatuurlijkPersoonDto,
                     public gehuwdOp: Date | null,
                     public gehuwdTe: string | null,
                     public kinderen: NatuurlijkPersoonDto[] | null,
                     public uitElkaar: Boolean | null) {
  }
}

export class RelatieIdDto {
  public constructor(public id: number) {
  }
}
